atom/common/native_mate_converters/callback.h

```diff
@@ -72,8 +72,14 @@ struct V8FunctionInvoker<ReturnType(ArgTypes...)> {
     v8::Context::Scope context_scope(context);
     ReturnType ret;
     std::vector<v8::Local<v8::Value>> args = { ConvertToV8(isolate, raw)... };
-    v8::Local<v8::Value> val(holder->Call(holder, args.size(), &args.front()));
-    Converter<ReturnType>::FromV8(isolate, val, &ret);
+    v8::Local<v8::Value> result;
+    auto maybe_result =
+        holder->Call(context, holder, args.size(), &args.front());
+    if (!maybe_result.ToLocal(&result)) {
+      static ReturnType defaultType;
+      return defaultType;
```

Comment:
It is calling `ReturnType`'s default constructor, the requirement of a default constructor has always been there.

Usually `ReturnType ret` is enough for most types, but for built-in types like `int` it results in an uninitialized state and we have to force using default constructor.